data_quality/tasks/extract_relevance_period.py

```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import datetime
+import dateparser
+from data_quality import compat
+from .base_task import Task
+from .check_datapackage import DataPackageChecker
+
+class RelevancePeriodExtractor(Task):
+
+    """A Task runner that extracts the period a sources's content reffers to
+        (is relevant for).
+    """
+
+    def __init__(self, config):
+        super(RelevancePeriodExtractor, self).__init__(config)
+        timeliness_params = self.config['timeliness']
+        self.timeliness_strategy = timeliness_params.get('timeliness_strategy', [])
+        self.date_order = timeliness_params.get('date_order', 'DMY')
+        if not self.timeliness_strategy:
+            raise ValueError('You need to provide values for "timeliness_strategy."')
+        datapackage_check = DataPackageChecker(self.config)
+        datapackage_check.check_database_completeness([self.source_file])
+        settings = {'RETURN_AS_TIMEZONE_AWARE': False,
+                    'PREFER_DAY_OF_MONTH': 'last',
+                    'PREFER_DATES_FROM': 'past',
+                    'SKIP_TOKENS': ['to'],
+                    'DATE_ORDER': self.date_order}
+        self.date_parser = dateparser.date.DateDataParser(settings=settings)
+
+    def run(self):
+        """Try to indentify a period based on timeliness strategy"""
+
+        sources = []
+        with compat.UnicodeDictReader(self.source_file) as sources_file:
+            for row in sources_file:
+                sources.append({field: val for field, val in row.items()
+                                if field in self.timeliness_strategy})
+
+        found_fields = set(sources[0].keys())
+        if not found_fields:
+            raise ValueError(('At least one of the "timeliness_strategy" fields '
+                              'must be present in your "sources_file".'))
+        if not found_fields.issuperset(set(self.timeliness_strategy)):
+            missing_fields = set(self.timeliness_strategy).difference(found_fields)
+            print(('Fields "{0}" from "timeliness_strategy" were not found '
+                   'in your `source_file`').format(missing_fields))
+
+        for source in sources:
+            extracted_period = self.resolve_period(source)
+            if not extracted_period:
+                period = ''
+            else:
+                period_start, period_end = extracted_period
+                period = '/'.join([compat.str(period_start.date()),
+                                   compat.str(period_end.date())])
+            source['period_id'] = period
+        return sources
```

Comment:
Not sure what to do here. Normally, we would like to write the updated sources in `source_file` overwriting the existing `period_id`. However, that might be bad for users who have their own `period_id`. Maybe we should add another configuration options `extract_period: true/false`?